kamu-adapter-http/src/ws_client_smart_transfer_protocol.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+use dill::component;
+use futures::SinkExt;
+
+use tokio_stream::StreamExt;
+use tokio_tungstenite::{
+    connect_async,
+    tungstenite::{Message},
+};
+
+use kamu::infra::utils::smart_transfer_protocol::SmartTransferProtocolClient;
+use kamu::domain::{Dataset, SyncError, SyncResult, SyncListener};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+
+#[component(pub)]
+pub struct WsSmartTransferProtocolClient {}
+
+#[async_trait::async_trait]
+impl SmartTransferProtocolClient for WsSmartTransferProtocolClient {
```

Comment:
name of the file doesn't correspond to to the name of the class - should it be `ws_smart_transfer_protocol_client.rs`?

Reply:
Renamed